_rules/SC4-1-2-aria-attr-valid.md

```diff
@@ -0,0 +1,114 @@
+---
+name: ARIA attribute is valid
+
+description: |
+ This rule checks that each aria- attribute specified is defined in ARIA 1.1
+
+success_criterion:
+- 4.1.2
+
+test_aspects:
+- DOM Tree
+
+authors:
+- Jey Nandakumar
+---
+
+## Test procedure
+
+### Applicability
+
+Any attribute that starts with `aria-`.
+
+### Expectation
+
+Each target attribute is defined in [WAI ARIA 1.1](https://www.w3.org/TR/wai-aria-1.1/).
+
+## Assumptions
+
+This rule does not cater to any custom `aria-` attribute that does not fall under the [WAI ARIA 1.1](https://www.w3.org/TR/wai-aria-1.1/) specification.
```

Comment:
If you provide a specification in the applicability, you will not need this here because you already defined it to exclude custom 'aria-' attributes.

Reply:
I'm not sure if I understand this assumption. The rule applies to 'aria-' attributes and checks if they are in the specification. The assumption seems to say that the 'aria-' attributes that are not in the specification are not applicable. Isn't this contradictory?

Reply:
The idea behind this assumption is to handle the edge case, where in custom attributes may be defined and scripted upon.

It would not see it as contradictory, as we emphasise `custom`.

Reply:
@jkodu so this would mean that if a developer uses consciously a custom aria- attribute then he may ignore the outcome of this rule? Just wondering if we should condone this usage of attributes starting with aria-